File: examples/PHP/mdclient.php
<?php
/*
 * Majordomo Protocol client example
 * Uses the mdcli API to hide all MDP aspects
 * 
 * @author Ian Barber <ian(dot)barber(at)gmail(dot)com>
 */
include_once "mdcliapi.php";

$verbose = $_SERVER['argc'] > 1 && $_SERVER['argv'][1] == '-v';
$session = new MDCli("tcp://localhost:5555", $verbose);
for($count = 0; $count < 100000; $count++) {
	$request = new Zmsg(); 
	$request->body_set("Hello world");
	$reply = $session->send("echo", $request);
	if(!$reply) {
		break; // Interrupt or failure
	}
}
printf ("%d requests/replies processed", $count);
echo PHP_EOL;

File: examples/Python/mdclient.py
"""
Majordomo Protocol client example. Uses the mdcli API to hide all MDP aspects

Author : Min RK <benjaminrk@gmail.com>

"""

import sys
from mdcliapi import MajorDomoClient

def main():
    verbose = '-v' in sys.argv
    client = MajorDomoClient("tcp://localhost:5555", verbose)
    count = 0
    while count < 100000:
        request = "Hello world"
        try:
            reply = client.send("echo", request)
        except KeyboardInterrupt:
            break
        else:
            # also break on failure to reply:
            if reply is None:
                break
        count += 1
    print "%i requests/replies processed" % count
    
if __name__ == '__main__':
    main()

